spec/imsc-hrm.html

```diff
@@ -449,40 +449,41 @@ <h2>Architecture</h2>
       <h2>General</h2>
 
       <p>The Presentation Compositor SHALL render in Presentation Buffer P<sub>n</sub> each successive <a>Intermediate Synchronic
-      Document</a> E<sub>n</sub> using the following steps in order:</p>
+      Document</a> E<sub>n</sub> that is not an <a>empty ISD</a> using the following steps in order:</p>
 
       <ol>
-        <li>clear the pixels of the entire <a>Root Container Region</a>, unless n=0 or E<sub>n-1</sub> is an <a>empty ISD</a>;</li>
-
         <li>paint, according to stacking order, all background pixels for each region;</li>
 
-        <li>paint all pixels for background colors associated with text or image subtitle content; and</li>
+        <li>paint all pixels for background colors associated with text or image subtitle content;</li>
+
+        <li>paint the text or image subtitle content; and</li>
 
-        <li>paint the text or image subtitle content.</li>
+        <li>clear the pixels of the entire <a>Root Container Region</a>.</li>
```

Comment:
This seems wrong: by moving this step to the end the result is always that all the pixels have been cleared, so all that painting we just did is pointless! Maybe it means clearing the pixels in a different buffer? Or something else?